src/Aspire.Hosting/Dashboard/CommandsConfigurationExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal static class CommandsConfigurationExtensions
+{
+    internal static IResourceBuilder<T> WithLifeCycleCommands<T>(this IResourceBuilder<T> builder) where T : IResource
```

Comment:
This name is probably noo generic.

Reply:
It's internal so it doesn't matter. Suggestions welcome.